add support to isRevoked check function

rrequero opened the pull request:
I have added the revoked feature. When the users want to check if the token is revoked, is possible with this new feature.

Thanks.

sc0ttyd commented on lib/index.js:
```diff
@@ -36,18 +36,28 @@ module.exports = function(opts) {
     }
 
+    let isRevoked = opts.isRevoked || (() => Promise.resolve());
+
```
If the revocation handler is moved into a named function, and the `let isRevoked ..` line is slightly refactored, then we can prevent the performance hit of the creation of two resolved promises and the execution of the handler from happening for the 95% (completely evidence-less estimate 😀 ) of users that don't use this feature:

```
const revocationHandler = revoked => {
          if (revoked) {
            return Promise.reject(new Error('Revoked token'));
          }
          return Promise.resolve(user)
}

const isRevoked = opts.isRevoked
    ? user => opts.isRevoked(user).then(revocationHandler)
    : user => user;

return JWT.verifyAsync(token, secret, opts)
   .then(isRevoked)
   .then((user) => {
       ctx.state = ctx.state || {};
       ctx.state[opts.key] = user;
   })
   // ...
```

NB this is not tested 😀

sc0ttyd commented on test/test.js:
```diff
@@ -5,7 +5,7 @@ const Koa     = require('koa');
 const assert  = require('assert');
 const jwt     = require('jsonwebtoken');
-
+const Promise = require('bluebird');
```
Node 4 (which is the min specified supported version in our package.json) has native promises, which should be sufficient, rather than introducing a large dependency.

rrequero commented:
Hi,
Thanks @ sc0ttyd for your review. You have reason with the refactoring. I changed the feature and I added your change, but I had to add a small change in the revocationHandler function because I need the user when the token is not revoked, but with the change all ok :)

About the promises, is true that node 4 supports natively but the library uses bluebird to promisify the jsonwebtoken library.

Thank very much

sc0ttyd commented:
Hi Raúl, you're welcome. The changes from the review look good to me. Still though, the tests work without test/tests.js line 8 (`const Promise = require('bluebird');`) 😀 

👍

sc0ttyd commented:
Hey man, you're quick 😀

rrequero commented:
hahahaha  😀

sc0ttyd commented:
Does anyone have an issue with this getting merged?
